src/main/java/org/terasology/climateConditions/ExtremeClimateConditionSystem.java

```diff
@@ -0,0 +1,198 @@
+                        entity.send(new DoDamageEvent(healthDecreaseAmount, frostbiteDamagePrefab));
+
+                            EntityRef particleEntity = entityManager.create("climateConditions:snowParticleEffect");
+                            LocationComponent targetLoc = entity.getComponent(LocationComponent.class);
+                            LocationComponent childLoc = particleEntity.getComponent(LocationComponent.class);
+                            childLoc.setWorldPosition(targetLoc.getWorldPosition());
+                            Location.attachChild(entity, particleEntity);
+                            particleEntity.setOwner(entity);
+                            Vector3f direction = targetLoc.getLocalDirection();
+                            direction.normalize();
+
+                            //NOTE: initializing velocity using the constructor wasn't working hence the long method.
+                            //particleEntity.addComponent(new VelocityRangeGeneratorComponent(direction,direction.scale(2)));
+                            VelocityRangeGeneratorComponent velocity = particleEntity.getComponent(VelocityRangeGeneratorComponent.class);
+                            direction.scale((float)0.5);
+                            direction.addY((float)0.5);
+                            velocity.minVelocity = direction;
+                            direction.scale((float)1.5);
+                            velocity.maxVelocity = direction;
+                            particleEntity.addOrSaveComponent(velocity);
+                            if(!entity.hasComponent(SnowParticleComponent.class)) {
+                                entity.addComponent(new SnowParticleComponent());
+                            }
+                            entity.getComponent(SnowParticleComponent.class).particleEntity = particleEntity;
+                    }
+            }
+        }
+    }
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_NORMAL + 10 )
+    public void onHealthRegen(ActivateRegenEvent event, EntityRef player, LocationComponent location) {
+        float height = location.getLocalPosition().getY();
+        if(height > thresholdHeight){
+            player.send(new DeactivateRegenEvent());
+        }
+    }
+    @ReceiveEvent
+    public void OnPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, CharacterMovementComponent movement) {
+        defaultRunFactor = movement.runFactor;;
+        defaultSpeedMultiplier = movement.speedMultiplier;
+        defaultJumpSpeed = movement.jumpSpeed;
+    }
+
+    @ReceiveEvent
+    public void OnSpeedReduce(ReduceSpeedEvent event, EntityRef player, CharacterMovementComponent movement){
+        movement.runFactor *= reducedRunFactorMultiplier;
+        movement.speedMultiplier *= reducedSpeedMultiplierMultiplier;
+        movement.jumpSpeed *= reducedJumpSpeedMultiplier;
```

Comment:
I'm not exactly sure, but it might be that you need to save the component back to the entity for it to take effect...

Reply:
Also, instead of modifying the values directly, we should consider introducing events similar to the character scale events.

Reply:
Will do that while incorporating the AlterationEffects module.